### PR TITLE
[SR-3481] Fix for memory leak in NSURLComponents 

### PR DESCRIPTION
**Pull request by nethraravindran**

### CODE REVIEW

File: Foundation/NSURL.swift [hunk 1]
```diff
@@ -948,8 +948,15 @@ open class NSURLQueryItem : NSObject, NSSecureCoding, NSCopying {
 }
 
 open class NSURLComponents: NSObject, NSCopying {
+    private let _cfinfo = _CFInfo(typeID: _CFURLComponentsGetTypeID())
     private let _components : CFURLComponentsRef!
     
+     deinit {
+        if let component = _components {
+            __CFURLComponentsDeallocate(component)
```
Comment (nethraravindran) [inline Foundation/NSURL.swift]: Called CoreFoundation deallocate method to free the resources.

File: CoreFoundation/URL.subproj/CFURLComponents.c [hunk 2]
```diff
@@ -66,9 +66,8 @@ static CFStringRef __CFURLComponentsCopyDescription(CFTypeRef cf) {
     return CFSTR("A really nice CFURLComponents object");
 }
 
-static void __CFURLComponentsDeallocate(CFTypeRef cf) {
-    CFURLComponentsRef instance = (CFURLComponentsRef)cf;
-    __CFGenericValidateType(cf, _CFURLComponentsGetTypeID());
+CF_EXPORT void __CFURLComponentsDeallocate(CFURLComponentsRef instance) {
```
Comment (nethraravindran) [inline CoreFoundation/URL.subproj/CFURLComponents.c]: Used CF_EXPORT in-order to call from deinit

File: Foundation/NSURL.swift [hunk 3]
```diff
@@ -948,8 +948,15 @@ open class NSURLQueryItem : NSObject, NSSecureCoding, NSCopying {
 }
 
 open class NSURLComponents: NSObject, NSCopying {
+    private let _cfinfo = _CFInfo(typeID: _CFURLComponentsGetTypeID())
```
Comment (ianpartridge) [inline Foundation/NSURL.swift]: What's the purpose of this line?

Comment (nethraravindran) [inline Foundation/NSURL.swift]: Realised it is no longer required. Will remove.

File: CoreFoundation/URL.subproj/CFURLComponents.c (hunk 2 above)
Comment (parkera) [inline CoreFoundation/URL.subproj/CFURLComponents.c]: Pleaes make this `CF_SWIFT_EXPORT`

File: CoreFoundation/URL.subproj/CFURLComponents.h [hunk 4]
```diff
@@ -30,6 +30,8 @@ typedef struct __CFURLComponents *CFURLComponentsRef;
 
 CF_EXPORT CFTypeID _CFURLComponentsGetTypeID(void);
 
+CF_EXPORT void __CFURLComponentsDeallocate(CFURLComponentsRef);
```
Comment (parkera) [inline CoreFoundation/URL.subproj/CFURLComponents.h]: `CF_SWIFT_EXPORT` here too.

**Comment by nethraravindran** [comment]
@parkera Addressed the changes.

**Comment by parkera** [comment]
@swift-ci please test and merge

**Pull request merged**